src/java.base/linux/classes/jdk/internal/platform/cgroupv1/CgroupV1Subsystem.java

```diff
@@ -0,0 +1,490 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import jdk.internal.platform.CgroupSubsystem;
+import jdk.internal.platform.CgroupSubsystemController;
+import jdk.internal.platform.CgroupUtil;
+import jdk.internal.platform.CgroupV1Metrics;
+
+public class CgroupV1Subsystem implements CgroupSubsystem, CgroupV1Metrics {
+    private CgroupV1MemorySubSystemController memory;
+    private CgroupV1SubsystemController cpu;
+    private CgroupV1SubsystemController cpuacct;
+    private CgroupV1SubsystemController cpuset;
+    private CgroupV1SubsystemController blkio;
+    private boolean activeSubSystems;
+
+    private static final CgroupV1Subsystem INSTANCE = initSubSystem();
+
+    private static final String PROVIDER_NAME = "cgroupv1";
+
+    private CgroupV1Subsystem() {
+        activeSubSystems = false;
+    }
+
+    public static CgroupV1Subsystem getInstance() {
+        return INSTANCE;
+    }
+
+    private static CgroupV1Subsystem initSubSystem() {
+        CgroupV1Subsystem subsystem = new CgroupV1Subsystem();
+
+        /**
+         * Find the cgroup mount points for subsystems
+         * by reading /proc/self/mountinfo
+         *
+         * Example for docker MemorySubSystem subsystem:
+         * 219 214 0:29 /docker/7208cebd00fa5f2e342b1094f7bed87fa25661471a4637118e65f1c995be8a34 /sys/fs/cgroup/MemorySubSystem ro,nosuid,nodev,noexec,relatime - cgroup cgroup rw,MemorySubSystem
+         *
+         * Example for host:
+         * 34 28 0:29 / /sys/fs/cgroup/MemorySubSystem rw,nosuid,nodev,noexec,relatime shared:16 - cgroup cgroup rw,MemorySubSystem
+         */
+        try (Stream<String> lines =
+                CgroupUtil.readFilePrivileged(Paths.get("/proc/self/mountinfo"))) {
+
+            lines.filter(line -> line.contains(" - cgroup "))
+                 .map(line -> line.split(" "))
+                 .forEach(entry -> createSubSystemController(subsystem, entry));
+
+        } catch (IOException e) {
+            return null;
+        }
+
+        /**
+         * Read /proc/self/cgroup and map host mount point to
+         * local one via /proc/self/mountinfo content above
+         *
+         * Docker example:
+         * 5:memory:/docker/6558aed8fc662b194323ceab5b964f69cf36b3e8af877a14b80256e93aecb044
+         *
+         * Host example:
+         * 5:memory:/user.slice
+         *
+         * Construct a path to the process specific memory and cpuset
+         * cgroup directory.
+         *
+         * For a container running under Docker from memory example above
+         * the paths would be:
+         *
+         * /sys/fs/cgroup/memory
+         *
+         * For a Host from memory example above the path would be:
+         *
+         * /sys/fs/cgroup/memory/user.slice
+         *
+         */
+        try (Stream<String> lines =
+                CgroupUtil.readFilePrivileged(Paths.get("/proc/self/cgroup"))) {
+
+            lines.map(line -> line.split(":"))
```

Comment:
We are missing the patch of JDK-8272124 here. Old code was:
https://github.com/openjdk/jdk11u-dev/blob/194b66f6a72437345c458e747e24beb46352537a/src/java.base/linux/classes/jdk/internal/platform/cgroupv1/Metrics.java#L119..L122

Best to use the JDK 17u commit as a base for this:
See: https://github.com/openjdk/jdk17u/commit/f0028333d83c39b434c26f66a6a92e446e7047cd#diff-7d6f202c58b4205f19a9b6c657f4827e9ef132d7433a7bf906b5b640f9f7841fR197

Looking at this again, we should bring in the test changes of JDK-8272124 too (as we are now bringing in those tests with the cgroups v2 backport). It helps to verify this fix is present as well.